lang/c++/impl/parsing/JsonCodec.cc

```diff
@@ -470,30 +466,30 @@ class JsonEncoder : public Encoder {
     JsonHandler<F> handler_;
     P parser_;
 
-    void init(OutputStream &os);
-    void flush();
-    int64_t byteCount() const;
-    void encodeNull();
-    void encodeBool(bool b);
-    void encodeInt(int32_t i);
-    void encodeLong(int64_t l);
-    void encodeFloat(float f);
-    void encodeDouble(double d);
-    void encodeString(const std::string &s);
-    void encodeBytes(const uint8_t *bytes, size_t len);
-    void encodeFixed(const uint8_t *bytes, size_t len);
-    void encodeEnum(size_t e);
-    void arrayStart();
-    void arrayEnd();
-    void mapStart();
-    void mapEnd();
-    void setItemCount(size_t count);
-    void startItem();
-    void encodeUnionIndex(size_t e);
+    void init(OutputStream &os) override;
```

Comment:
if these are being declared in a .cc should they be final instead?

Reply:
Good point. Took care of this and similar uses throughout.

Reply:
also, not sure if they already are, but if these aren't, then enclosing private details in the anonymous namespace is also a good idea.